src/Dodo.HttpClient.ResiliencePolicies/ResiliencePoliciesSettings.cs

```diff
@@ -1,26 +1,100 @@
 using System;
+using System.Net.Http;
 using Dodo.HttpClientResiliencePolicies.CircuitBreakerPolicy;
 using Dodo.HttpClientResiliencePolicies.RetryPolicy;
 using Dodo.HttpClientResiliencePolicies.TimeoutPolicy;
+using Polly;
 
 namespace Dodo.HttpClientResiliencePolicies
 {
 	public class ResiliencePoliciesSettings
 	{
-		public ITimeoutPolicySettings OverallTimeoutPolicySettings { get; set; }
+		private TimeoutPolicySettings _overallTimeoutPolicySettings = new TimeoutPolicySettings();
+		private TimeoutPolicySettings _timeoutPerTryPolicySettings = new TimeoutPolicySettings();
+		private RetryPolicySettings _retryPolicySettings = new RetryPolicySettings();
+		private CircuitBreakerPolicySettings _circuitBreakerPolicySettings = new CircuitBreakerPolicySettings();
 
-		public ITimeoutPolicySettings TimeoutPerTryPolicySettings { get; set; }
+		public ResiliencePoliciesSettings()
+		{
+		}
+
+		public ResiliencePoliciesSettings(
+			TimeSpan overallTimeout,
+			TimeSpan timeoutPerTry,
+			RetryPolicySettings retryPolicyPolicySettings,
+			CircuitBreakerPolicySettings circuitBreakerPolicyPolicySettings)
```

Comment:
I thought about it. Here we have a trade off between usability and "correctness" I could say. If we add `OnBreak/OnReset/OnHalfOpen` back to the `CircuiteBreakerPolicySettings`, we have to choose:
1. Remove it from the root settings as it was before.
2. Deal with inconsistent state. I mean settings creation like this:

```csharp
var cbSettings = new CircuitBreakerPolicySettings
{
    OnBreak = myOnBreakHandler1,
};

var settings = new ResiliencePolicySettings
{
    OnBreak = myOnBreakHandler2,
    CircuitBreakerPolicySettings = cbSettings,
}
```
How should we choose between this two handlers? Last wins - could be confusing, because CB settings could be created somewhere in other place. Decorate CB OnBreak with root settings OnBreak? Possible, but overcomplicated in my opinion.

Reply:
I have one more idea about it. One of the major difference between our library and Polly is than Polly is like a Lego and your can build your own policies step-by-step choosing proper building block, but we provide a monolithic solution, you can tune it a bit, but you can't change the flow. In case of Polly it make sense to keep `OnRetry` action inside the `RetryPolicy` because you can use it alone without other staff, but in our case you can't choose use or do not use one of the inner policies. It means, that our `OnRetry` action it is actually top level action. You choice is not to subscribe or not subscribe to action of RetryPolicy, you choose subscribe or not subscribe to the Dodo.HttpClient.ResiliencePolicy OnRetry action. It is important difference.